### PR TITLE
Automatically deploy tagged releases to bucket

### PR DESCRIPTION
**Pull request by ashb**
This uses a new helper repo I created https://github.com/astronomer/pip-release-utilities -- as we have three or so modules deploying to there and we want to share some of the deploy steps.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -27,8 +49,47 @@ jobs:
...
+
+  publish:
+    docker:
+      - image: gcr.io/google.com/cloudsdktool/cloud-sdk:alpine
+    steps:
+      - checkout
+      - *restore_venv
+      - run:
+          name: verify tag
+          command: |-
```
Comment (andriisoldatenko) [inline .circleci/config.yml]: are you sure about `|-`?
from docs:
```
- run:
    command: |
      echo Running test
      mkdir -p /tmp/test-results
      make test
```

Comment (ashb) [reply .circleci/config.yml]: I thought `|` didn't chomp leading space and `|-` did, but yes either way this works - https://app.circleci.com/jobs/github/astronomer/astronomer-airflow-version-check/49

Comment (ashb) [reply .circleci/config.yml]: Ah https://yaml-multiline.info/ `-` is the Block Chomping Indicator` -- so makes no really difference in  this case as empty lines aren't significant to bash.

**Comment by ashb** [comment]
(BTW Andrii I was thinking we'd convert all our python modules related to Airflow to use this deploy pipepline.)

**Pull request merged**